src/fusion_power_plant.h

```diff
@@ -59,27 +56,173 @@ class FusionPowerPlant : public cyclus::Facility  {
+    "default": 1000.0, \
+    "doc": "Initial mass of full blanket material", \
+    "tooltip": "Only blanket material mass, not structural mass", \
+    "units": "kg", \
+    "uitype": "range", \
+    "range": [0, 10000], \
+    "uilabel": "Initial Mass of Blanket" \
+  }
+  double blanket_size;
+
+  #pragma cyclus var { \
+    "default": 0.05, \
+    "doc": "Percent of blanket that gets recycled every blanket turnover period", \
+    "tooltip": "Defaults to 0.05 (5%), must be between 0 and 15%", \
+    "units": "dimensionless", \
+    "uitype": "range", \
+    "range": [0, 0.15], \
+    "uilabel": "Blanket Turnover Rate" \
+  }
+  double blanket_turnover_rate;
+
+  //Resource Buffers and Trackers:
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_storage;
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> helium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_feed;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_waste;
 
-  //PyneIDs
-  const int tritium_id = 10030000;
+  cyclus::toolkit::MatlBuyPolicy fuel_startup_policy;
+  cyclus::toolkit::MatlBuyPolicy fuel_refill_policy;
+  cyclus::toolkit::MatlBuyPolicy blanket_fill_policy;
+
+  cyclus::toolkit::MatlSellPolicy tritium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy helium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy blanket_waste_sell_policy;
 
-  //Compositions:
-  const cyclus::CompMap T = {{tritium_id, 1}};
-  const cyclus::Composition::Ptr tritium_comp = cyclus::Composition::CreateFromAtom(T);
+  cyclus::toolkit::TotalInvTracker fuel_tracker;
+  cyclus::toolkit::TotalInvTracker blanket_tracker;
 
-  //Materials:
-  cyclus::Material::Ptr sequestered_tritium = cyclus::Material::CreateUntracked(0.0, tritium_comp);
+  //Functions:
+  void CycleBlanket();
+  bool CheckOpeartingConditions();
+  bool CheckInitialStartupCondition();
+  void SequesterTritium();
+  void OperateReactor();
+  void CycleBlanket();
+  void DecayInventories();
+  void ExtractHelium();
+  void MoveExcessTritiumToSellBuffer();
+
+  private:
+    double fuel_usage_atoms;
+    double fuel_usage_mass;
+    const double seconds_per_year = 2629846*12;
+    const double MW_to_GW = 1000.0;
+    double avogadros_number = pyne::N_A; //6.022e23;
+    double fuel_limit = 1000.0;
+    double blanket_limit = 100000.0; 
+
+
+    //NucIDs for Pyne
+    const int tritium_id = 10030000;
+    const int He3_id = 20030000;
+    const int He4_id = 20040000;
+    const int Li6_id = 30060000;
+    const int Li7_id = 30070000;
+
+    //masses in kg/atom
+    //Maybe make pyne do some of this...
+    const double amu_to_kg = 1.66054e-27;
+    const double Li7_atomic_mass = pyne::atomic_mass(Li7_id) * amu_to_kg;
+    const double Li6_atomic_mass = pyne::atomic_mass(Li6_id) * amu_to_kg;
+    const double tritium_atomic_mass = pyne::atomic_mass(tritium_id) 
+                 * amu_to_kg;
+    const double He4_atomic_mass = pyne::atomic_mass(He4_id) * amu_to_kg;
```

Comment:
Arguably, we don't need to define these until we are sure what we need as you write the algorithms elsewhere...  This is also what's driving the need for `amu_to_kg`

Reply:
Removed a lot of the code that was non-essential. Should I go ahead and fold Issue #15 into this PR so that you can see what (a trimmed down) EnterNotify() is doing, or should we leave that for the next one? I forget what we ultimately decided yesterday.

Reply:
I am interested in seeing the PR associated with #15 as it's own PR into `main` to see what it requires.